cloudnet/src/main/java/de/dytanic/cloudnet/command/commands/CommandMigrate.java

```diff
@@ -0,0 +1,119 @@
+
+            Integer chunkSize = Ints.tryParse(properties.getOrDefault("chunk-size", Integer.toString(100)));
+            if (chunkSize == null) {
+              chunkSize = 100;
+            }
+
+            for (String databaseName : sourceDatabaseProvider.getDatabaseNames()) {
+              sender.sendMessage(
+                LanguageManager.getMessage("command-migrate-current-database").replace("%db%", databaseName));
+
+              Database sourceDatabase = sourceDatabaseProvider.getDatabase(databaseName);
+              Database targetDatabase = targetDatabaseProvider.getDatabase(databaseName);
+
+              sourceDatabase.iterate(targetDatabase::insert, chunkSize);
+            }
+
+            executeIfNotCurrentProvider(sourceDatabaseProvider, AbstractDatabaseProvider::close);
+            executeIfNotCurrentProvider(targetDatabaseProvider, AbstractDatabaseProvider::close);
+
+            sender.sendMessage(LanguageManager.getMessage("command-migrate-success")
+              .replace("%source%", sourceDatabaseProvider.getName())
+              .replace("%target%", targetDatabaseProvider.getName()));
+          },
+          command -> command.enableProperties().appendUsage("| --no-clear --chunk-size=100"),
+          anyStringIgnoreCase("database", "db"),
+          dynamicString("database-from", LanguageManager.getMessage("command-migrate-unknown-database-provider"),
+            providerRegisteredPredicate(),
+            possibleAnswerSupplier()),
+          dynamicString("database-to", LanguageManager.getMessage("command-migrate-unknown-database-provider"),
+            providerRegisteredPredicate(),
+            possibleAnswerSupplier())
+        )
+        .getSubCommands(),
+      "migrate"
+    );
+    super.prefix = "cloudnet";
+    super.permission = "cloudnet.command." + super.names[0];
+    super.description = LanguageManager.getMessage("command-description-migrate");
+  }
+
+  private static Predicate<String> providerRegisteredPredicate() {
+    return input -> getDatabaseProvider(input) != null;
+  }
+
+  private static Supplier<Collection<String>> possibleAnswerSupplier() {
+    return () -> CloudNet.getInstance().getServicesRegistry().getServices(AbstractDatabaseProvider.class)
+      .stream()
+      .map(INameable::getName)
+      .collect(Collectors.toList());
+  }
+
+  private static AbstractDatabaseProvider getDatabaseProvider(@NotNull String name) {
+    return CloudNet.getInstance().getServicesRegistry().getService(AbstractDatabaseProvider.class, name);
+  }
+
+  private static void executeIfNotCurrentProvider(@NotNull AbstractDatabaseProvider provider,
+    @NotNull ThrowableConsumer<AbstractDatabaseProvider, ?> handler) {
+    if (!CloudNet.getInstance().getDatabaseProvider().equals(provider)) {
+      try {
+        handler.accept(provider);
+      } catch (Throwable throwable) {
+        throwable.printStackTrace();
```

Comment:
Don’t catch Throwables, only catch Exceptions as Errors would get catched too. Also this should be properly logged or rethrown.